bubble_sort.rb

```diff
@@ -0,0 +1,39 @@
+# bubble_sort method
+def bubble_sort(array)
+  len = array.length
+  (0...len - 1).each do |i|
+    (0...len - 1 - i).each do |j|
+      array[j], array[j + 1] = array[j + 1], array[j] if array[j] > array[j + 1]
+    end
+  end
+end
+
+# bubble_sort_by method
+def bubble_sort_by(array)
+  len = array.length - 1
+  swapped = true
+  while swapped
+    swapped = false
+    (0..len - 1).each do |i|
+      next unless yield(array[i], array[i + 1]).positive?
+
+      swap = array[i + 1]
+      array[i + 1] = array[i]
+      array[i] = swap
+      swapped = true
+    end
+    len -= 1
+  end
+  p array
+end
```

Comment:
- [ ] This is a logic method, so rather than printing the value, you should return the value instead. This is also to ensure that your method passes the automated tests used for testing your method. 
If you need to see the output of your method, you can call a print statement when the method is called.
Also, remember that Ruby methods automatically returns the last evaluated statement,  therefore, there is no need for explicitly writing the return statement. You can take advantage of that.